opay_8.1.gateway.class.v1.2.15.php

```diff
@@ -349,279 +325,255 @@ function autoAlign(){
-                  ."Accept-Encoding: identity\r\n" // this client does not support a compression
-                  ."Accept-Charset: utf-8;q=0.7,*;q=0.7\r\n";
+            . "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\n"
+            . "Accept-Language: en-us,en;q=0.5\r\n"
+            . "Accept-Encoding: identity\r\n" // this client does not support a compression
+            . "Accept-Charset: utf-8;q=0.7,*;q=0.7\r\n";
         
-        if ($httpMethod == 'GET')
-        {
+        if ($httpMethod === 'GET') {
             $url = strtok($url, '?');
-            if (isset($parsedUrlArr['query']))
-            {
+
+            if (isset($parsedUrlArr['query'])) {
                 parse_str($parsedUrlArr['query'], $queryParamsArr);
-                if (!empty($queryParamsArr))
-                {
+                if (!empty($queryParamsArr)) {
                     $parametersArray = array_merge($queryParamsArr, $parametersArray);    
                 }
             }
-            $url .= '?'.http_build_query($parametersArray, '', '&', PHP_QUERY_RFC1738);    
-        }
-        else
-        {
-            $headers .= "Content-Length: ".strlen($content)."\r\n";
-        }
 
-        if (!$keepAlive)
-        {
-            $headers .= "Connection: Close\r\n";
-        }
-        else
-        {
-            $headers .= "Connection: keep-alive\r\n";
-        }
-           
-        if ($optionalHeaders !== null) {
-            $headers .= $optionalHeaders;
+            $url .= '?' . http_build_query($parametersArray, '', '&', PHP_QUERY_RFC1738);
+        } else {
+            $headers .= "Content-Length: " . strlen($content)."\r\n";
         }
+
+        $headers .= !$keepAlive ? "Connection: Close\r\n" : "Connection: keep-alive\r\n";
+        $headers .= $optionalHeaders !== null ? $optionalHeaders : '';
         
-        // removing the last \r\n for possible incompatibilities in some conjunctions of PHP + OpenSSL when file_get_contents returns === false
+        // Remove the last \r\n for possible incompatibilities in some conjunctions of PHP + OpenSSL when file_get_contents returns === false
         $headers = rtrim($headers);
-        
+
+        if (function_exists('curl_init')) {
+            try {
+                $ch = curl_init();
+                curl_setopt($ch, CURLOPT_URL, $url);
+                if ($httpMethod !== 'GET') {
+                    curl_setopt($ch, CURLOPT_POST, 1);
+                    curl_setopt($ch, CURLOPT_POSTFIELDS, $content);
+                }
+                curl_setopt($ch, CURLOPT_HTTPHEADER, explode("\r\n", $headers));
+                curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+                curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $timeout);
+                $data = curl_exec($ch);
+                curl_close($ch);
+
+                return $data;
+            } catch (Exception $e) {
+                // Move to the next method
+            }
+        }
```

Comment:
cURL doesn't throw exceptions as far as I know. curl_exec returns false on failure. You can do something like this.
```suggestion
        if (function_exists('curl_init')) {
            $ch = curl_init();
            curl_setopt($ch, CURLOPT_URL, $url);
            
            if ($httpMethod !== 'GET') {
                curl_setopt($ch, CURLOPT_POST, 1);
                curl_setopt($ch, CURLOPT_POSTFIELDS, $content);
            }
            
            curl_setopt($ch, CURLOPT_HTTPHEADER, explode("\r\n", $headers));
            curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
            curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, $timeout);
            $data = curl_exec($ch);
            curl_close($ch);

            if ($data !== false) {
                return $data;
            }
        }
```

Reply:
Added additional if, but left try-catch to be sure file_get_contents() will be called if cURL fails because of any other reasons.